platformics/codegen/templates/database/migrations/env.py.j2

```diff
@@ -43,6 +43,8 @@ def run_migrations_offline() -> None:
     )
 
     with context.begin_transaction():
+        context.get_context()._ensure_version_table()  # pylint: disable=protected-access
+        connection.execute(sa.sql.text("LOCK TABLE alembic_version IN ACCESS EXCLUSIVE MODE"))
```

Comment:
This ensures that if multiple processes (e.g. k8s pods) try to run migrations at the same time, only one of them will actually be able to run them.